src/DotNet.Status.Web/TimelineIssueTriageService.cs

```diff
@@ -0,0 +1,286 @@
+        private readonly ZenHubClient _zenHub;
+        private readonly TimelineIssueTriageLogic _logic;
+
+        public TimelineIssueTriageService(
+            IGitHubApplicationClientFactory gitHubApplicationClientFactory,
+            IKustoIngestClientFactory clientFactory,
+            IOptions<KustoOptions> kustoOptions,
+            IOptions<GitHubConnectionOptions> githubOptions,
+            ZenHubClient zenHub,
+            ILogger<TimelineIssueTriageService> logger)
+        {
+            _logger = logger;
+            _gitHubApplicationClientFactory = gitHubApplicationClientFactory;
+            _kustoIngestClientFactory = clientFactory;
+            _kustoOptions = kustoOptions;
+            _githubOptions = githubOptions;
+            _zenHub = zenHub;
+            _logic = new TimelineIssueTriageLogic();
+        }
+
+        public async Task ProcessIssueEvent(IssuesHookData issuePayload)
+        {
+            if (issuePayload.Action != "opened" &&
+                issuePayload.Action != "reopened" &&
+                issuePayload.Action != "closed")
+            {
+                _logger.LogInformation($"Received github action '{issuePayload.Action}', nothing to do");
+                return;
+            }
+
+            // Determine identifiable information for triage items
+            var triageItems = GetTriageItemsProperties(issuePayload.Issue.Body);
+
+            if (!triageItems.Any())
+            {
+                /* Item is not a triage item (does not contain identifiable information), do nothing */
+                _logger.LogInformation($"{issuePayload.Issue.Url} is not a triage type issue.");
+
+                return;
+            }
+
+            IGitHubClient gitHubClient = await _gitHubApplicationClientFactory.CreateGitHubClientAsync(issuePayload.Repository.Owner.Login, issuePayload.Repository.Name);
+
+            string updatedCategory = null;
+
+            if (issuePayload.Action == "opened" || issuePayload.Action == "reopened")
+            {
+                // First, look for duplicate issues that are open
+                var openIssues = new RepositoryIssueRequest
+                {
+                    Filter = IssueFilter.All,
+                    State = ItemStateFilter.Open,
+                    SortProperty = IssueSort.Created,
+                    SortDirection = SortDirection.Ascending,
+                };
+                openIssues.Labels.Add(_labelName);
+
+                _logger.LogInformation("Getting open issues");
+                var existingTriageIssues = await gitHubClient.Issue.GetAllForRepository(issuePayload.Repository.Id, openIssues);
+                _logger.LogInformation($"There are {existingTriageIssues.Count} open issues with the '{_labelName}' label");
+                foreach (var existingIssue in existingTriageIssues)
+                {
+                    if (existingIssue.Number != issuePayload.Issue.Number)
```

Comment:
nit: I think it's a bit more redable to do `==` and `continue;` then go lower, do the same with `!IsDuplicate()` until you reach the logic

Reply:
It could be also written as LINQ query:
```
var existingDuplicate = existingTriageIssues.FirstOrDefault(i =>
    i.Number != issuePayload.Issue.Number
    && IsDuplicate(triageItems, GetTriageItemsProperties(existingIssue.Body)));

if (existingDuplicate != null)
{
    ...
    return;
}
```

Reply:
This code was copy pasted. We made decision to not refactor it all